dao/mongo.go

```diff
@@ -585,6 +585,52 @@ func (m *MongoService) CreateResolutionResource(dao *models.ResolutionResourceDa
 	return http.StatusCreated, nil
 }
 
+// CreateStatementOfAffairsResource stores the statement of affairs resource for the insolvency case
+// with the specified transactionID
+func (m *MongoService) CreateStatementOfAffairsResource(dao *models.StatementOfAffairsResourceDao, transactionID string) (int, error) {
+	var insolvencyResource models.InsolvencyResourceDao
+	collection := m.db.Collection(m.CollectionName)
+
+	filter := bson.M{"transaction_id": transactionID}
+
+	statementDao := models.StatementOfAffairsResourceDao{
+		StatementDate: dao.StatementDate,
+		Attachments:   dao.Attachments,
+	}
+
+	// Retrieve insolvency case from Mongo
+	storedInsolvency := collection.FindOne(context.Background(), filter)
+	err := storedInsolvency.Err()
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			log.Debug("no insolvency resource found for transaction id", log.Data{"transaction_id": transactionID})
+			return http.StatusNotFound, fmt.Errorf("there was a problem handling your request for transaction %s not found", transactionID)
+		}
+		log.Error(err)
+		return http.StatusInternalServerError, fmt.Errorf("there was a problem handling your request for transaction %s", transactionID)
+	}
+
+	err = storedInsolvency.Decode(&insolvencyResource)
+	if err != nil {
+		log.Error(err)
+		return http.StatusInternalServerError, fmt.Errorf("there was a problem handling your request for transaction %s", transactionID)
+	}
+
+	update := bson.M{
```

Comment:
If a user were to create another SOA after already having created one, would the original one be overwritten?

Reply:
Yes, I've spoken with our analyst about this and it is expected behaviour